### DISCUSSION TITLE
get [Errno 2] No such file or directory: 'python'  with code  'runner.run() '

### DISCUSSION THREAD

**Discussion by lg2578** [Project-MONAI/tutorials#1200] [Q&A]
I'm try auto3dseg in my machine. and get the following error  with runner.run():
2023-01-30 11:29:25,353 - INFO - Skipping data analysis...
2023-01-30 11:29:25,355 - INFO - Skipping algorithm generation...
2023-01-30 11:29:25,362 - INFO - Launching: python /mnt/spleenreg/dints_0/scripts/search.py run --config_file='/mnt/spleenreg/dints_0/configs/network.yaml','/mnt/spleenreg/dints_0/configs/transforms_validate.yaml','/mnt/spleenreg/dints_0/configs/transforms_infer.yaml','/mnt/spleenreg/dints_0/configs/network_search.yaml','/mnt/spleenreg/dints_0/configs/hyper_parameters.yaml','/mnt/spleenreg/dints_0/configs/transforms_train.yaml','/mnt/spleenreg/dints_0/configs/hyper_parameters_search.yaml' --searching#num_iterations=8 --searching#num_iterations_per_validation=4 --searching#num_images_per_batch=2 --searching#num_epochs=2 --searching#num_warmup_iterations=4
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
/tmp/ipykernel_106/3926302587.py in <cell line: 1>()
----> 1 runner.run()

~/miniconda3/envs/myconda/lib/python3.8/site-packages/monai/apps/auto3dseg/auto_runner.py in run(self)
    681 
    682             if not self.hpo:
--> 683                 self._train_algo_in_sequence(history)
    684             else:
    685                 self._train_algo_in_nni(history)

~/miniconda3/envs/myconda/lib/python3.8/site-packages/monai/apps/auto3dseg/auto_runner.py in _train_algo_in_sequence(self, history)
    553         for task in history:
    554             for _, algo in task.items():
--> 555                 algo.train(self.train_params)
    556                 acc = algo.get_score()
    557                 algo_to_pickle(algo, template_path=algo.template_path, best_metrics=acc)

/mnt/spleenreg/algorithm_templates/dints/scripts/algo.py in train(self, train_params)
    396         cmd, devices_info = self._create_cmd(dints_search_params)
    397         cmd_search = cmd.replace("train.py", "search.py")
--> 398         self._run_cmd(cmd_search, devices_info)
    399 
    400         # training

~/miniconda3/envs/myconda/lib/python3.8/site-packages/monai/apps/auto3dseg/bundle_gen.py in _run_cmd(self, cmd, devices_info)
    186         if devices_info:
    187             ps_environ["CUDA_VISIBLE_DEVICES"] = devices_info
--> 188         normal_out = subprocess.run(cmd.split(), env=ps_environ, check=True)
    189 
    190         return normal_out

~/miniconda3/envs/myconda/lib/python3.8/subprocess.py in run(input, capture_output, timeout, check, *popenargs, **kwargs)
    491         kwargs['stderr'] = PIPE
    492 
--> 493     with Popen(*popenargs, **kwargs) as process:
    494         try:
    495             stdout, stderr = process.communicate(input, timeout=timeout)

~/miniconda3/envs/myconda/lib/python3.8/subprocess.py in __init__(self, args, bufsize, executable, stdin, stdout, stderr, preexec_fn, close_fds, shell, cwd, env, universal_newlines, startupinfo, creationflags, restore_signals, start_new_session, pass_fds, encoding, errors, text)
    856                             encoding=encoding, errors=errors)
    857 
--> 858             self._execute_child(args, executable, preexec_fn, close_fds,
    859                                 pass_fds, cwd, env,
    860                                 startupinfo, creationflags, shell,

~/miniconda3/envs/myconda/lib/python3.8/subprocess.py in _execute_child(self, args, executable, preexec_fn, close_fds, pass_fds, cwd, env, startupinfo, creationflags, shell, p2cread, p2cwrite, c2pread, c2pwrite, errread, errwrite, restore_signals, start_new_session)
   1702                     if errno_num != 0:
   1703                         err_msg = os.strerror(errno_num)
-> 1704                     raise child_exception_type(errno_num, err_msg, err_filename)
   1705                 raise child_exception_type(err_msg)
   1706 

FileNotFoundError: [Errno 2] No such file or directory: 'python'

**Comment by lg2578** [comment]
I tried to print the cmd in algo.py,and copy it to terminal ,also get errors
2023-01-30 11:49:16,242 - INFO - Skipping data analysis...
2023-01-30 11:49:16,244 - INFO - Skipping algorithm generation...
python /mnt/spleenreg/dints_0/scripts/train.py run --config_file='/mnt/spleenreg/dints_0/configs/network.yaml','/mnt/spleenreg/dints_0/configs/transforms_validate.yaml','/mnt/spleenreg/dints_0/configs/transforms_infer.yaml','/mnt/spleenreg/dints_0/configs/network_search.yaml','/mnt/spleenreg/dints_0/configs/hyper_parameters.yaml','/mnt/spleenreg/dints_0/configs/transforms_train.yaml','/mnt/spleenreg/dints_0/configs/hyper_parameters_search.yaml' --searching#num_iterations=8 --searching#num_iterations_per_validation=4 --searching#num_images_per_batch=2 --searching#num_epochs=2 --searching#num_warmup_iterations=4
***
python /mnt/spleenreg/dints_0/scripts/search.py run --config_file='/mnt/spleenreg/dints_0/configs/network.yaml','/mnt/spleenreg/dints_0/configs/transforms_validate.yaml','/mnt/spleenreg/dints_0/configs/transforms_infer.yaml','/mnt/spleenreg/dints_0/configs/network_search.yaml','/mnt/spleenreg/dints_0/configs/hyper_parameters.yaml','/mnt/spleenreg/dints_0/configs/transforms_train.yaml','/mnt/spleenreg/dints_0/configs/hyper_parameters_search.yaml' --searching#num_iterations=8 --searching#num_iterations_per_validation=4 --searching#num_images_per_batch=2 --searching#num_epochs=2 --searching#num_warmup_iterations=4 0
2023-01-30 11:49:16,251 - INFO - Launching: python /mnt/spleenreg/dints_0/scripts/search.py run --config_file='/mnt/spleenreg/dints_0/configs/network.yaml','/mnt/spleenreg/dints_0/configs/transforms_validate.yaml','/mnt/spleenreg/dints_0/configs/transforms_infer.yaml','/mnt/spleenreg/dints_0/configs/network_search.yaml','/mnt/spleenreg/dints_0/configs/hyper_parameters.yaml','/mnt/spleenreg/dints_0/configs/transforms_train.yaml','/mnt/spleenreg/dints_0/configs/hyper_parameters_search.yaml' --searching#num_iterations=8 --searching#num_iterations_per_validation=4 --searching#num_images_per_batch=2 --searching#num_epochs=2 --searching#num_warmup_iterations=4

### errors in terminal:
[Errno 2] No such file or directory: '/mnt/spleenreg/dints_0/configs/network.yaml,/mnt/spleenreg/dints_0/configs/transforms_validate.yaml,/mnt/spleenreg/dints_0/configs/transforms_infer.yaml,/mnt/spleenreg/dints_0/configs/network_search.yaml,/mnt/spleenreg/dints_0/configs/hyper_parameters.yaml,/mnt/spleenreg/dints_0/configs/transforms_train.yaml,/mnt/spleenreg/dints_0/configs/hyper_parameters_search.yaml'

**Comment by KumoLiu** [comment]
Hi @lg2578, could you please try this command and check whether you are under the correct path. You can find some guidance under "...../dints_0/docs)"
```
python -m scripts.search run --config_file "['configs/hyper_parameters.yaml','configs/hyper_parameters_search.yaml','configs/network_search.yaml','configs/transforms_train.yaml','configs/transforms_validate.yaml']"
```

Hope it can help you, thanks!

**Reply by lg2578** [reply]
your code  works in the .../dints_0 directory.
I changed the code to format like yours. and it works.
python scripts/search.py run --config_file "['configs/network.yaml','configs/transforms_validate.yaml','configs/transforms_infer.yaml','configs/network_search.yaml','configs/hyper_parameters.yaml','configs/transforms_train.yaml','configs/hyper_parameters_search.yaml']" --searching#num_iterations=8 --searching#num_iterations_per_validation=4 --searching#num_images_per_batch=2 --searching#num_epochs=2 --searching#num_warmup_iterations=4 0
it seems format error. --config_file=..... should be config_file "[,,,]"
